light_ws2812_AVR/Light_WS2812/light_ws2812.c

```diff
@@ -133,17 +134,30 @@ void inline ws2812_sendarray_mask(uint8_t *data,uint16_t datlen,uint8_t maskhi)
   ws2812_DDRREG |= maskhi; // Enable output
   
   masklo	=~maskhi&ws2812_PORTREG;
+  // `masklo` becomes the value to write to the PORT? register in order to
+  // drive the DATA line low.
   maskhi |=        ws2812_PORTREG;
+  // `maskhi` becomes the value to write to the PORT? register in order to
+  // drive the DATA line high.  All other bits in the PORT? register are
+  // preserved by becoming part of the byte that is written here.
   
   sreg_prev=SREG;
+
+  // If we are not expected to support interrupts while driving the LED
+  // string..
 #ifdef interrupt_is_disabled
   cli();  
 #endif  
 
+  // Go through each byte that should be sent to the LED string..
   while (datlen--) {
     curbyte=*data++;
     
+#ifndef __GNUC__
```

Comment:
I think this is not necessary, as the __asm__ keyword also works when the c99 std is not used.